src/lib/hooks/use-page-title.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+
+/**
+ * Grab the page `title`, and update it when the route changes.
+ *
+ * Note: this is intended for use in announcing route changes.
+ * Generally, it would be preferable to grab the string that we use to
+ * set the page title directly, rather than extracting it from the DOM.
+ */
+export default function usePageTitle() {
+	const [pageTitle, setPageTitle] = useState('')
+	const router = useRouter()
+
+	useEffect(() => {
+		const updatePageTitle = () => {
+			setPageTitle(document.title)
+		}
+		updatePageTitle()
```

Comment:
@zchsh it could be announcing twice because you are calling it here and then it also fires on the event `routeChangeComplete`.

Reply:
Could be! I find that when I remove this, I only get intermittent announcements, and sometimes the page title announced is incorrect:

https://github.com/hashicorp/dev-portal/assets/4624598/48ab6183-e75e-49d5-86b8-703f4fe15bd4

Have tried a few other things but can't seem to get things to consistently announce only once on route change... 

Confusingly, the NextJS docs claim that [Next should announce client-side page transitions by default](https://nextjs.org/docs/architecture/accessibility#route-announcements). That doesn't seem to be what we're seeing here... so maybe another path forward could be to reach out to the NextJS team to get support on this... I'll reach out tomorrow!